compiler-rt/lib/profile/InstrProfilingFile.c

```diff
@@ -923,6 +925,9 @@ static void parseAndSetFilename(const char *FilenamePat,
       PROF_NOTE("Override old profile path \"%s\" via %s to \"%s\" via %s.\n",
                 OldFilenamePat, getPNSStr(OldPNS), lprofCurFilename.FilenamePat,
                 getPNSStr(PNS));
+
+    free(OldFilenamePat);
```

Comment:
Wouldn't this new allocation be leaked if we return in either of the two returns above, between the alloc and this free?

Reply:
Right, I recall checking that this was not the case within the first early return, but definitely would be in the second one, thanks. Releasing the allocation on both of them.